components/ws-manager/pkg/manager/status.go

```diff
@@ -716,8 +717,15 @@ func (m *Manager) isWorkspaceTimedOut(wso workspaceObjects) (reason string, err
 		return decide(start, m.Config.Timeouts.TotalStartup, activity)
 
 	case api.WorkspacePhase_RUNNING:
-		timeout := m.Config.Timeouts.RegularWorkspace
-		activity := activityNone
+		// First check is always for the max lifetime
+		timeout := m.Config.Timeouts.MaxLifetime
+		activity := activityMaxLifetime
+		if msg, err := decide(start, timeout, activity); msg != "" {
+			return msg, err
+		}
```

Comment:
nit: can be simplified (we don't need the variable assignment here because there's no variance in the decision)
```suggestion
		if msg, err := decide(start, m.Config.Timeouts.MaxLifetime, activityMaxLifetime); msg != "" {
			return msg, err
		}
```